src/requirements.h

```diff
@@ -190,6 +210,8 @@ struct requirement_data {
         const requirement_data disassembly_requirements() const;
 
     private:
+        std::string id_;
```

Comment:
I wonder why this is a plain string and not a `requirement_id`. Same for the parameter of `save_requirement`/`load_requirement`. They are used like ids, they are even named `id`, why doesn't the type match?